### ISSUE TITLE
Import error if type "module" is specified in package.json

### ISSUE THREAD

**Issue by ACertainCoder** [ivandotv/radio-browser-api#59]
Node.js throws this error if you try to import radio-browser-api and the project type is set to "module" in the package.json:
```
(node:11516) Warning: To load an ES module, set "type": "module" in the package.json or use the .mjs extension.
(Use `node --trace-warnings ...` to show where the warning was created)
C:\...\node_modules\radio-browser-api\dist\esm\index.js:1
export * from './radioBrowser';
^^^^^^
```
https://nodejs.org/api/packages.html#type

Would it be possible to change the file extension to mjs or something to fix this issue?

**Comment by ivandotv** [comment]
As far as I'm aware I can't have both `cjs` and `mjs` extensions generated from the same source code.
Do you have any suggestions?

helpful references:
https://nodejs.org/api/packages.html#determining-module-system
https://nodejs.org/api/packages.html#dual-commonjses-module-packages

Although I might drop the commonjs version all together now the Node v15 is LTS

**Comment by ACertainCoder** [comment]
I've done some research and it seems like the TypeScript compiler cannot output .mjs or .cjs files out of the box...

If you drop the support for commonjs entirely, the easiest thing would be to simply add `"type": "module"` to the `package.json`, which tells Node.js that this is an `esm` module alltogether.

But I also found this example repository: https://github.com/gfmio/typescript-esm-cjs-hybrid-example - it seems to support both, `cjs` and `esm` modules and solves the mentioned issue above by renaming the files after the build process is completed. I haven't tried it myself yet but it looks promising.

**Comment by ivandotv** [comment]
@ACertainCoder  
I've switched the build process to microbundle and added the `module` export.
Can you try it? Just download it from npm

**Comment by ACertainCoder** [comment]
Hi, I've tested the current version (5.2.3) but sadly still get the same error:
`(node:6544) Warning: To load an ES module, set "type": "module" in the package.json or use the .mjs extension.`

```
SyntaxError: Unexpected token 'export'
    at Object.compileFunction (node:vm:352:18)
    at wrapSafe (node:internal/modules/cjs/loader:1032:15)
    at Module._compile (node:internal/modules/cjs/loader:1067:27)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1155:10)
    at Module.load (node:internal/modules/cjs/loader:981:32)
    at Function.Module._load (node:internal/modules/cjs/loader:822:12)
    at ModuleWrap.<anonymous> (node:internal/modules/esm/translators:168:29)
    at ModuleJob.run (node:internal/modules/esm/module_job:195:25)
    at async Promise.all (index 0)
    at async ESMLoader.import (node:internal/modules/esm/loader:337:24)
    at async loadESM (node:internal/process/esm_loader:88:5)
    at async handleMainPromise (node:internal/modules/run_main:61:12)
```

**Comment by ACertainCoder** [comment]
My current workaround is to manually change the file extension to "mjs" and adjust the package.json to the following:
```json
  "main": "./dist/prod/radio-browser-api.js",
  "module": "./dist/prod/radio-browser-api.esm.mjs",
  "umd": "./dist/prod/radio-browser-api.umd.js",
  "exports": {
    "development": "./dist/dev/radio-browser-api.esm.mjs",
    "import": "./dist/prod/radio-browser-api.esm.mjs",
    "require": "./dist/prod/radio-browser-api.js"
  }
```

Would it be possible to export the file with this extension or maybe rename it afterwards?

**Comment by ivandotv** [comment]
@ACertainCoder I've changed the export to something similar. I think it works now. You can download the latest version from `npm`.

https://github.com/ivandotv/radio-browser-api/blob/682f02d01d9b4e050f3a847641b08fb56bed61ad/package.json#L14-L21

**Comment by ACertainCoder** [comment]
Seems like it works! 🥳 Node.js was able to start my project without any issues.

**Comment by ivandotv** [comment]
closed via ca49d5d5174b517ad3d73c662e260ca67f8a4ee3
